src/test/java/it/gov/pagopa/pu/bff/util/DateUtilsTest.java

```diff
@@ -83,4 +89,63 @@ void whenOnlyToDateIsSetThenThrowException() {
 
     assertTrue(actualMessage.contains(expectedMessage));
   }
+
+  @Test
+  void testToOffsetDateTimeEndOfTheDay() {
+    OffsetDateTime expected = LocalDateTime.of(LocalDate.now(),
+        LocalTime.MAX.truncatedTo(java.time.temporal.ChronoUnit.MILLIS))
+      .atZone(Constants.ZONEID).toOffsetDateTime();
+
+    OffsetDateTime result = DateUtils.toOffsetDateTimeEndOfTheDay(
+      LocalDate.now());
+
+    assertConversion(expected, result);
+  }
+
+  @Test
+  void testToOffsetDateTimeStartOfTheDay() {
+    OffsetDateTime expected = LocalDate.now().atStartOfDay(Constants.ZONEID).toOffsetDateTime();
+
+    OffsetDateTime result = DateUtils.toOffsetDateTimeStartOfTheDay(
+      LocalDate.now());
+
+    assertConversion(expected, result);
+  }
+
+  @Test
+  void givenNullLocalDateWhenToOffsetDateTimeEndOfTheDayThenReturnNull() {
+    assertNull(DateUtils.toOffsetDateTimeStartOfTheDay(null));
+  }
+
+  @Test
+  void givenNullLocalDateWhenToOffsetDateTimeStartOfTheDayThenReturnNull() {
+    assertNull(DateUtils.toOffsetDateTimeStartOfTheDay(null));
+  }
+
+  @Test
+  void testToRangeClosedOffsetDateTimeIntervalFilter() {
+    OffsetDateTime from = LocalDate.now().atStartOfDay(Constants.ZONEID).toOffsetDateTime();
+    OffsetDateTime to = LocalDateTime.of(LocalDate.now(),
+        LocalTime.MAX.truncatedTo(java.time.temporal.ChronoUnit.MILLIS))
+      .atZone(Constants.ZONEID).toOffsetDateTime();
+
+    OffsetDateTimeIntervalFilter expected = new OffsetDateTimeIntervalFilter(from, to);
+
+    it.gov.pagopa.pu.processexecutions.dto.generated.OffsetDateTimeIntervalFilter result = DateUtils.toRangeClosedOffsetDateTimeIntervalFilter(
+      new LocalDateIntervalFilter(LocalDate.now(), LocalDate.now()));
+
+    assertNotNull(result);
+    assertEquals(expected.getFrom(), result.getFrom());
+    assertEquals(expected.getTo(), result.getTo());
+  }
+
+  private static void assertConversion(OffsetDateTime expected, OffsetDateTime result) {
+    assertNotNull(result);
+    assertEquals(expected.getYear(), result.getYear());
+    assertEquals(expected.getMonthValue(), result.getMonthValue());
+    assertEquals(expected.getDayOfMonth(), result.getDayOfMonth());
+    assertEquals(expected.getHour(), result.getHour());
+    assertEquals(expected.getMinute(), result.getMinute());
+    assertEquals(expected.getSecond(), result.getSecond());
+  }
```

Comment:
Just use Assertions.assertTrue(expected.isEqual(result)), you don't even need to create a method to do that...